project/src/main/webapp/emailParse.js

```diff
@@ -0,0 +1,98 @@
+// limitations under the License.
+/** Class to load ML Model and find the best answer using email bodies. */
+class MLModelEmailParse {
+    constructor() {
+        this.model;
+        this.loadModel();
+    }
+
+    loadModel() {
+        //let modelLoadReady = document.getElementById("modelLoad");
+        //modelLoadReady.innerHTML = 'Loading Model...';
+        console.time("Model Load");
+        qna.load().then(loadedModel => {
+            this.model = loadedModel;
+            console.timeEnd("Model Load");
+            //modelLoadReady.innerHTML = 'Model Ready';
+        });
+    }
+
+    /** Parse Email Bodies to apply ML Model using Promises */
+    parseEmailsWithModel() {
+        // class from Gmail API js file
+        const gmail = new gmailAPI();
+        const {question, totalObjects} = gmail;
+        const allPass = this.extractEmailBodiesToArray(totalObjects);
+        console.time("Using Promises Test");
+        const promises = allPass.map(
+            passage => this.model.findAnswers(question, passage),
+        );
+        Promise.all(promises).then(values => {
+            const nonEmpty = this.getScoreToEmail(values, allPass);
+            const answer = document.getElementById("answer");
+            if (nonEmpty.size == 0) {
+                answer.innerHTML = "No Answer Available";
+            }
+            else {
+                const orderedConfidence = this.highestConfidence(nonEmpty);
+                const mlDictAnswer = nonEmpty.get(orderedConfidence);
+                console.log("original", values);
+                console.log("nonEmpty", nonEmpty);
+                console.log("highest confidence", orderedConfidence);
+                answer.innerHTML = mlDictAnswer["answer"];
+                console.timeEnd("Using Promises Test");
+            }
+        });
+    }
+
+    extractEmailBodiesToArray(exampleDict) {
+        const allPass = [];
+        for (let key in exampleDict) {
+            let body = exampleDict[key]["emailBody"];
+            allPass.push(body.slice(300));
+        }
+        return allPass;
+    }
+
+    /** Create map of scores with values as ML answer and email body */
+    getScoreToEmail(mlValues, allPass) {
+        const confidenceWithEmailBody = new Map();
+        for (let i = 0; i < mlValues.length; i++) {
+            if (mlValues[i].length != 0) {
+                let mlAnswer = mlValues[i][0]["text"];
+                confidenceWithEmailBody.set(mlValues[i][0]["score"], { "answer": mlAnswer, "emailBody": allPass[i] });
+            }
+        }
+        return confidenceWithEmailBody;
+    }
+
+    highestConfidence(answerDict) {
+        let confidenceArray = [];
+        for (let key of answerDict.keys()) {
+            confidenceArray.push(key);
+        }
+        let bestAnswer = confidenceArray.sort((a, b) => b - a);
+        return bestAnswer[0];
+    }
+}
+
+let mlClass = new MLModelEmailParse();
```

Comment:
Try putting this in window.onload to fix the race with loading the model in the constructor:

```
window.onload = {
  const mlClass = new MLModelEmailParse();
}
```

http://go/mdn/API/GlobalEventHandlers/onload